internal/pkg/service/buffer/worker/conditions/checker.go

```diff
@@ -0,0 +1,336 @@
+// Package conditions provides the Checker to periodically check file import and slice upload conditions.
+package conditions
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/benbjohnson/clock"
+	"github.com/keboola/go-client/pkg/client"
+	"github.com/keboola/go-client/pkg/storageapi"
+	etcd "go.etcd.io/etcd/client/v3"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/log"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/statistics"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/storage/file"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/store"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/store/key"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/store/model"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/store/schema"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/worker/distribution"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/worker/task"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/etcdop"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/rollback"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/servicectx"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
+)
+
+const (
+	CheckInterval                = 30 * time.Second
+	MinimalCredentialsExpiration = time.Hour
+)
+
+type Checker struct {
+	clock            clock.Clock
+	logger           log.Logger
+	schema           *schema.Schema
+	etcdClient       *etcd.Client
+	httpClient       client.Client
+	storageAPIHost   string
+	store            *store.Store
+	tasks            *task.Node
+	dist             *distribution.Node
+	statsCache       *statistics.CacheNode
+	uploadConditions model.Conditions
+}
+
+type conditionsMap map[key.ExportKey]model.Conditions
+
+type slicesMap map[key.SliceKey]sliceInfo
+
+// expiration of the slice upload credentials.
+type sliceInfo struct {
+	expiration *time.Time
+}
+
+type dependencies interface {
+	Clock() clock.Clock
+	Logger() log.Logger
+	Process() *servicectx.Process
+	Schema() *schema.Schema
+	EtcdClient() *etcd.Client
+	HTTPClient() client.Client
+	StorageAPIHost() string
+	Store() *store.Store
+	TaskWorkerNode() *task.Node
+	DistributionWorkerNode() *distribution.Node
+	StatsCacheNode() *statistics.CacheNode
+}
+
+func NewChecker(d dependencies) (*Checker, error) {
+	v := &Checker{
+		clock:            d.Clock(),
+		logger:           d.Logger().AddPrefix("[conditions]"),
```

Comment:
Implementacia kontroly podmienok, ... kod by sa dal este upravit aby bol krajsi, ale teraz to neriesim.